scipy/optimize/_minpack_py.py

```diff
@@ -620,6 +621,18 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         `mesg`, and `ier`.
 
         .. versionadded:: 1.9
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan.
+        The following options are available (default is 'propagate'):
+
+          * 'propagate': just propagate nan values.
+          * 'raise': throws an error
+          * 'omit': performs the calculations ignoring nan values
+
+        Note that if you want to use `omit` or `propagate` without the finite
+        checks, `check_finite` should be set to False.
```

Comment:
We already have `check_finite` argument which is covered by `nan_policy`, but for backward compatibility, I think we should keep it and `nan_policy` works only when `check_finite` is False.

Reply:
Another possibility for backward compatibility is to have the default be `nan_policy=None`, in which case the function will behave exactly as it does now, and no NaN checks will even be performed. But if the user specifies `nan_policy`, the behavior could override (or work together with) the `check_finite`. I want to add `nan_policy=None` to stats, so I think this would still be consistent (considering the direction that stats is going).